clang/lib/CodeGen/CGAtomic.cpp

```diff
@@ -1241,7 +1245,8 @@ RValue CodeGenFunction::EmitAtomicExpr(AtomicExpr *E) {
 
     }
     // Optimized functions have the size in their name.
-    // XXXAR: for capability libcalls we use _cap as the suffix
+    // XXXAR: for capability libcalls we use _cap as the suffix to avoid calling
+    // the _8/_16 version that does not know about capabilities.
```

Comment:
Not sure that's an XXX TBH; an atomic operation on __int128 on architectures that support that _shouldn't_ preserve the tag.

Reply:
Will update the comment. I just wanted to state that using the _16 call is *wrong* for capabilities.